### PR TITLE
Np 7221 search organization name

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/project.feature [hunk 1]
```diff
@@ -88,7 +88,7 @@ Feature: API tests for Cristin Project retrieve and search
     Then status 400
     And match response.title == 'Bad Request'
     And match response.status == 400
-    And match response.detail == "Invalid query parameter supplied. Valid parameter is 'language'"
+    And match response.detail == "Invalid query parameter supplied. Valid parameter(s) is 'language'"
```
Comment (brinxmat) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/project.feature:91]: What about "Valid parameters: …"?, thus avoiding any issues with formatting?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personQuery.feature [hunk 2]
```diff
@@ -98,7 +99,7 @@ Feature: API tests for Cristin persons query
     And match response.title == 'Bad Request'
     And match response.status == 400
     # TODO: Change detail to exclude language
```
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personQuery.feature:101]: Can remove todo now

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java [hunk 3]
```diff
@@ -41,31 +42,38 @@ protected String getValidPage(RequestInfo requestInfo) throws BadRequestExceptio
...
                 .map(UriUtils::escapeWhiteSpace)
-                .orElseThrow(() -> new BadRequestException(ERROR_MESSAGE_NAME_MISSING_OR_HAS_ILLEGAL_CHARACTERS));
+                .orElseThrow(() -> new BadRequestException(
+                        invalidQueryParametersMessage(NAME, ALPHANUMERIC_CHARACTERS_DASH_COMMA_PERIOD_AND_WHITESPACE)));
     }
 
+    protected Optional<String> getValidOrganization(RequestInfo requestInfo) {
+        return getQueryParameter(requestInfo, ORGANIZATION)
+                .filter(this::isValidQueryString)
+                .map(UriUtils::escapeWhiteSpace);
```
Comment (torarnet) [inline cristin-commons/src/main/java/no/unit/nva/cristin/common/handler/CristinQueryHandler.java:75]: Do this escapeWhiteSpace change "Norwegian Business School" to "NorwegianBusinessSchool"?

**Review by torarnet — DISMISSED** [review]

File: cristin-commons/src/main/java/no/unit/nva/cristin/common/ErrorMessages.java [hunk 4]
```diff
@@ -2,57 +2,66 @@
...
-        "The requested resource '%s' was not found";
+            "The requested resource '%s' was not found";
     public static final String ERROR_MESSAGE_INVALID_PAYLOAD = "Supplied payload is not valid";
     public static final String ERROR_MESSAGE_INVALID_PERSON_ID = "Invalid path parameter for person id";
 
+    public static String validQueryParametersMessage(Set<String> queryParameters) {
+        return String.format(ERROR_MESSAGE_TEMPLATE_INVALID_QUERY_PARAMETERS, prettifyList(queryParameters));
+    }
+
+    public static String invalidQueryParametersMessage(String queryParameterName, String validValues) {
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply cristin-commons/src/main/java/no/unit/nva/cristin/common/ErrorMessages.java:66]: Some javadoc added

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personQuery.feature (hunk 2 above)
Comment (svenngjeto) [reply cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personQuery.feature]: done

**Review by torarnet — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**